### ISSUE TITLE
Feature Request: Loading weights for layers defined in tf.layers api

### ISSUE THREAD

**Issue by AakashKumarNain** [tensorflow/tensorflow#12746]
Let's say I define a layer using the `tf.layers` API as shown below:

```
conv1 = tf.layers.conv2d(input_img, filters=32, 
                                     kernel_size=(3,3), 
                                     padding='same', 
                                     name='Conv1')
```

Now I can build a whole network defining such layers. Can you please introduce another functionality for the `tf.layers` api so that for each layer we can set the weights in a single line like this:

`conv1.set_weights(weights)` or something like this `conv1.set_params(param_values)`

This would be very very useful.

**Comment by martinwicke** [comment]
If you use the Object oriented layers (on master), you can access the variables directly:

```
conv1 = tf.layers.Conv2d(...)
tf.assign(conv1.kernel, [1,2,3,4]) 
```

(also check out `init_from_checkpoint`)

Would these work in your use case?

**Comment by AakashKumarNain** [comment]
Correct me if I am wrong, wouldn't it be required to reshape the weights as per the previous layer output shape? Doing this check for each layer in the net would be a mess. With the feature that I requested for, I think that this check can be included in the method itself where the output of incoming layer is checked, the weights are reshaped accordingly and then assigned to the kernel

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by martinwicke** [comment]
If you have to reshape weights, they will most likely not work. You either have weights of the correct size, then they can be loaded using init_from_checkpoint, or you don't, in which case, it is highly unlikely to work at all.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by AakashKumarNain** [comment]
Why should I go for a complicated way? The high-level api should come with such methods. Every framework which is a high level api, for example keras or lasagne, comes with such a nice way to set the weights for each layer independently

**Comment by martinwicke** [comment]
All layers in tf.keras come with this ability. You should use them.

I still believe that there would never be a need to reshape, or if there was, you'd want to know.
